MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioFactory.java

```diff
@@ -295,12 +284,9 @@ public static void createScenariosForNewWeek(Campaign c) {
                         } else {
                             // edge case: lance assigned to another mission gets assigned the scenario,
                             // we need to remove any scenario they are assigned to already
-                            for (Scenario scenario : c.getMission(lance.getMissionId()).getScenarios()) {
-                                if ((scenario instanceof AtBScenario)
-                                        && (((AtBScenario) scenario).getLanceForceId() == lance.getForceId())) {
-                                    c.getMission(lance.getMissionId()).removeScenario(scenario.getId());
-                                }
-                            }
+                            c.getMission(lance.getMissionId()).getScenarios().removeIf(scenario ->
```

Comment:
One point against using anonymous functions like this is that they're harder to debug if the logic inside is non-trivial. Not necessarily saying you need to change it, but something to keep in mind.